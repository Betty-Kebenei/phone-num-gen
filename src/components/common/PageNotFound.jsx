import React from 'react';
import '../../styles/PageNotFound.css';

const PageNotFound = (props) => (
  <div className="page-not-found">
    <p className="page-text">Page Not Found</p>
    <p className="status-text">404</p>
    <p className="go-text">GO TO</p>
    <button
      id="generate"
      type="button"
      onClick={() => props.history.push('/')}
    >
      Generate Numbers
    </button>
    <span className="or-text"> OR </span>
    <button
      id="view"
      type="button"
      onClick={() => props.history.push('/numbers')}
    >
      View All Numbers
    </button>
  </div>
);
export default PageNotFound;
