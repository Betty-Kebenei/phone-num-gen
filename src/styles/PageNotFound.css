.page-not-found {
  text-align: center;
  margin-top: 10vh;
}

.page-text {
  font-size: 7vw;
  margin: 0;
  color: #007bff;
}

.status-text {
  font-size: 20vw;
  margin: 0;
}

button {
  border-radius: 5px;
  padding: 5px;
  background: #007bff;
  color: #fff;
}
