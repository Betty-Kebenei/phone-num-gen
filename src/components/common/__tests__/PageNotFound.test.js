import React from 'react';
import renderer from 'react-test-renderer';
import {configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import PageNotFound from '../PageNotFound';

configure({ adapter: new Adapter() })
global.window = { location: { pathname: null } };

const props = {
  history: {
    push: jest.fn()
  }
}

describe('PageNotFound', () => {
  test('page not found renders correctly', () => {
    const component = renderer.create(<PageNotFound />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('buttons', () => {
    test('Generate Numbers Button', () => {
      const wrapper = mount(<PageNotFound {...props} />);
      wrapper.find('#generate').simulate('click');
      expect(global.window.location.pathname).toEqual('/');
    })

    test('View All Numbers Button', () => {
      const wrapper = mount(<PageNotFound {...props} />);
      wrapper.find('#view').simulate('click');
      // expect(global.window.location.pathname).toEqual('/numbers');
    })
  })
})
