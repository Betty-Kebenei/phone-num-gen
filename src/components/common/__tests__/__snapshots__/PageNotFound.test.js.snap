// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageNotFound page not found renders correctly 1`] = `
<div
  className="page-not-found"
>
  <p
    className="page-text"
  >
    Page Not Found
  </p>
  <p
    className="status-text"
  >
    404
  </p>
  <p
    className="go-text"
  >
    GO TO
  </p>
  <button
    id="generate"
    onClick={[Function]}
    type="button"
  >
    Generate Numbers
  </button>
  <span
    className="or-text"
  >
     OR 
  </span>
  <button
    id="view"
    onClick={[Function]}
    type="button"
  >
    View All Numbers
  </button>
</div>
`;
